File: 0112-path-sum/0112-path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void helper(int sum, boolean[] found, TreeNode root, int targetSum)
    {
        if(root == null)
            return;
        sum += root.val;
        if(root.left == null && root.right == null && sum == targetSum)
        {
            found[0] = true;
            return;
        }
        helper(sum, found, root.left, targetSum);
        helper(sum, found, root.right, targetSum);
    }
    public boolean hasPathSum(TreeNode root, int targetSum) {
        boolean found[] = new boolean[1];
        found[0] = false;
        helper(0, found, root, targetSum);
        return found[0];
    }
}

File: 0112-path-sum/0112-path-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(TreeNode* root, int target, int sum, bool& found){
        if(!root) return;
        sum += root->val;
        if(root->left == NULL && root->right == NULL && sum == target){
            found = true;
            return;
        }
        helper(root->left, target, sum, found);
        helper(root->right, target, sum, found);
    }
    bool hasPathSum(TreeNode* root, int targetSum) {
        bool found = 0;
        helper(root, targetSum, 0, found);
        return found;
    }
};